tests-web/www/js/main.js

```diff
@@ -1,7 +1,230 @@
+    this.layers = layers;
+    this.bounds = bounds;
+  }
+
+  cleanup() {
+    for (const layer of this.layers) {
+      this.app.removeLayer(layer);
+    }
+  }
+
+  static async loadFromServer(app, name) {
+    const prefix = `tests/${name}/`;
+    const osmInput = await loadFile(prefix + "input.osm");
+    const rawMap = loadFile(prefix + "raw_map.json");
+    const network = loadFile(prefix + "road_network.dot");
+
+    const rawMapLayer = makePlainGeoJsonLayer(await rawMap);
+    const bounds = rawMapLayer.getBounds();
+
+    var layers = [];
+    layers.push(app.addLayer("Geometry", rawMapLayer));
+    layers.push(app.addLayer("OSM", makeOsmLayer(osmInput)));
+    layers.push(
+      app.addLayer("Network", await makeDotLayer(await network, { bounds }))
+    );
+
+    const drivingSide = JSON.parse(await loadFile(prefix + "test.json"))[
+      "driving_side"
+    ];
+
+    return new TestCase(app, name, osmInput, drivingSide, layers, bounds);
+  }
+
+  static async importCurrentView(app, importButton) {
+    // Grab OSM XML from Overpass
+    // (Sadly toBBoxString doesn't seem to match the order for Overpass)
+    const b = app.map.getBounds();
+    const bbox = `${b.getSouth()},${b.getWest()},${b.getNorth()},${b.getEast()}`;
+    const query = `(nwr(${bbox}); node(w)->.x; <;); out meta;`;
+    const url = `https://overpass-api.de/api/interpreter?data=${query}`;
+    console.log(`Fetching from overpass: ${url}`);
+
+    importButton.innerText = "Downloading from Overpass...";
+    // Prevent this function from happening twice in a row. It could also maybe
+    // be nice to allow cancellation.
+    importButton.disabled = true;
+
+    // TODO If this fails, I guess app.currentTest winds up null?
+    try {
+      const resp = await fetch(url);
+      const osmInput = await resp.text();
+
+      importButton.innerText = "Importing OSM data...";
+
+      // TODO Ask overpass
+      const drivingSide = "Right";
+
+      const network = new JsStreetNetwork(osmInput, {
+        driving_side: drivingSide,
+      });
+      const rawMapLayer = makePlainGeoJsonLayer(network.toGeojsonPlain());
+      const bounds = rawMapLayer.getBounds();
+
+      var layers = [];
+      layers.push(app.addLayer("Geometry", rawMapLayer));
+      layers.push(
+        app.addLayer(
+          "Detailed geometry",
+          makeDetailedGeoJsonLayer(network.toGeojsonDetailed())
+        )
+      );
+      layers.push(app.addLayer("OSM", makeOsmLayer(osmInput)));
+      // TODO The dot layer
+
+      return new TestCase(app, null, osmInput, drivingSide, layers, bounds);
+    } catch (err) {
+      window.alert(`Import failed: ${err}`);
+    } finally {
+      // Make the button clickable again
+      importButton.innerText = "Import current view";
+      importButton.disabled = false;
+    }
+  }
+
+  renderControls(container) {
+    container.innerHTML = "";
+    if (this.name) {
+      const title = container.appendChild(document.createElement("b"));
```

Comment:
The visual output here is weird; I was expecting each child of the `ul` to wind up on its own line.

Also this feels like a clunky API for creating new elements like this. Any suggested improvements without reaching for a huge framework?